src/mapobjects.py

```diff
@@ -0,0 +1,84 @@
+import os.path
+from dataclasses import dataclass
+from xml.etree import ElementTree
+
+import pygame
+
+from src.enums import Tileset
+from src.settings import SCALE_FACTOR
+
+
+@dataclass
+class MapObjectType:
+    id: int = None
+
+    image: pygame.Surface = None
+    hitbox: pygame.FRect = None
+
+
+class MapObjects:
+    objects: dict[int, MapObjectType]
```

Comment:
This might be just a nitpick, but why not use a list or tuple for this, if you are going to use regular ints as keys?

Reply:
I wanted to map the exact ID of the Tile in the Tileset to the corresponding image and hitbox. It is still a considerable option, but I think we should stick to dictionaries just in case the tileset is not formatted perfectly.

Reply:
Normally, tiles in Tiled tilesets are ordered when created, and if an ID is deleted it should be the first used to create a new tile afterwards. In case an ID doesn't exist you can just put None in the list and add a check for that

Reply:
I would still prefer to stick with the dictionary.
At the moment there are already no objects with IDs 1-8 and 48-55. I just had a look and it looks like new tiles in objects.tsx start with IDs 48+, and new tiles in trees and bushes.tsx with IDs 78+, so I don't think that the now unused IDs will be reassigned too quickly.
Also I think that the dictionary approach could be more efficient as we could only load objects from the current map without having to go through the IDs of all objects used on all maps.